lib/rubocop/cop/style/stabby_lambda_parentheses.rb

```diff
@@ -0,0 +1,92 @@
+# encoding: utf-8
+module RuboCop
+  module Cop
+    module Style
+      # Check for parentheses around stabby lambda arguments
+      class StabbyLambdaParentheses < Cop
+        include ConfigurableEnforcedStyle
+
+        def on_send(node)
+          return unless valid_node?(node)
+
+          if style == :require_parentheses
+            if !parentheses?(node)
+              missing_parentheses(node)
+            else
+              correct_style_detected
+            end
+          elsif parentheses?(node)
+            unwanted_parentheses(node)
+          else
+            correct_style_detected
+          end
+        end
+
+        def autocorrect(node)
+          if style == :require_parentheses
+            missing_parentheses_corrector(node)
+          elsif style == :require_no_parentheses
+            unwanted_parentheses_corrector(node)
+          end
+        end
+
+        def missing_parentheses(node)
+          add_offense(node_args(node), :expression,
+                      'Use with parentheses when there are parameters.') do
+            opposite_style_detected
+          end
+        end
+
+        def unwanted_parentheses(node)
+          add_offense(node_args(node), :expression,
+                      'Use without parentheses when there are parameters.') do
+            opposite_style_detected
+          end
+        end
+
+        private
+
+        def missing_parentheses_corrector(node)
+          lambda do |corrector|
+            args_loc = node_args(node).loc.expression
+
+            corrector.insert_before(args_loc, '(')
+            corrector.insert_after(args_loc, ')')
+          end
+        end
+
+        def unwanted_parentheses_corrector(node)
+          lambda do |corrector|
+            args_loc = node_args(node).loc
+
+            corrector.replace(args_loc.begin, '')
+            corrector.remove(args_loc.end)
+          end
+        end
+
+        def valid_node?(node)
+          lambda_node?(node) && args?(node)
+        end
+
+        def lambda_node?(node)
+          _, call = node.to_a
+          call == :lambda
+        end
+
+        def node_args(node)
+          node.parent.children.detect(&:args_type?)
+        end
+
+        def args?(node)
+          args = node_args(node)
+          args.children.count > 0
```

Comment:
I was mostly referring to the fact that the `children` are the actual arguments. I can live with the current version of the code.